File: src/Modelo/Material.java
package Modelo;

import java.io.Serializable;

public class Material implements Serializable{

    private static final long serialVersionUID = 1L;
    private String nome;
    private String link;

    public Material(String nome, String link){
        this.nome = nome;
        this.link = link;
    }

    public String obterNome(){
        return nome;
    }

    public String obterLink(){
        return link;
    }

    @Override
    public String toString(){
        return nome + ": " + link;
    }
}

File: cpp_version/Modelo/Material.cpp
#include "Material.h"

Material :: Material (string nome, string link){
    this->nome = nome;
    this->link = link;
}

string Material :: obterNome(){
    return nome;
}

string Material :: obterLink(){
    return link;
}

ostream& operator<<(ostream& os, Material &material){
    os << material.nome << ": " << material.link << endl;
    return os;
}